api/chisel/tunnel.go

```diff
@@ -38,26 +37,33 @@ func randomInt(min, max int) int {
 	return min + rand.Intn(max-min)
 }
 
-// GetTunnelDetails returns information about the tunnel associated to an environment(endpoint).
-func (service *Service) GetTunnelDetails(endpointID portainer.EndpointID) *portainer.TunnelDetails {
+// NOTE: it needs to be called with the lock acquired
```

Comment:
what about adding something like https://github.com/trailofbits/go-mutexasserts to check this?

Reply:
Interesting, I'll create a proposal after doing some necessary changes and this could be a part of it.